compiler/rustc_monomorphize/src/collector.rs

```diff
@@ -257,8 +257,16 @@ pub fn collect_crate_mono_items(
 ) -> (FxHashSet<MonoItem<'_>>, UsageMap<'_>) {
     let _prof_timer = tcx.prof.generic_activity("monomorphization_collector");
 
-    let roots =
-        tcx.sess.time("monomorphization_collector_root_collections", || collect_roots(tcx, mode));
+    let (roots, _) = join(
+        || {
+            tcx.sess
+                .time("monomorphization_collector_root_collections", || collect_roots(tcx, mode))
+        },
+        || {
+            // Prefetch upstream_monomorphizations
```

Comment:
Could you elaborate why?

Reply:
I added an more extended comment here. I also made it more conditional. I still expect it to regress for crates which use no upstream generics in debug mode, as `upstream_monomorphizations` wouldn't be needed for those.